### PR TITLE
[DO NOT MERGE] Paradise by the dashboard light

### PR DESCRIPTION
**Pull request by adambraimbridge**
cc @wheresrhys: This can wait for Monday but it's starting to look okay

Oh and I'm not married to the term "postulate", but it does help to have a word to define the object that contains the alias, name, queryname, query and other properties. I propose to replace "alias" — it doesn't have to be "postulate", but it kind of fits, considering it's defined as "a thing suggested or assumed as true as the basis for reasoning, discussion, or belief." 

... perhaps we should just call them "charts"

### CODE REVIEW

**Comment by adambraimbridge** [comment]
cc @wheresrhys: This can wait for Monday but it's starting to look okay

**Comment by adambraimbridge** [comment]
image

File: server/controllers/dashboard.js [hunk 1]
```diff
@@ -4,21 +4,49 @@
...
+
+	const dashboardname = req.params[0].split('/')[0];
+
+	// find dashboardname in res.locals.dashboards
+	let dashboard = res.locals.dashboards.filter(d => {
+		return d.id === dashboardname;
+	})[0];
+
+	// Append from the spreadsheet of destiny
+	dashboard.postulates = dashboard.postulates || [];
```
Comment (wheresrhys) [inline server/controllers/dashboard.js]: I'm not a fan of postulates to be honest, but `charts` would be a good move. The spreadsheet originally drove a CLI, so alias made sense then, but not any more.

File: server/middleware/dashboards/anonymous.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+# A postulate is "a thing suggested or assumed as true as the basis for reasoning, discussion, or belief."
+postulates:
+  -
+    question: "Are daily anonymous visitor counts increasing?"
+    dashboardfeature: "anonymous"
+    queryname: "daily"
+    name: "anonymous/daily"
+    query: "page:view->count(user.ab.allocationId)->filter(!user.uuid)"
+    printer: null
```
Comment (wheresrhys) [inline server/middleware/dashboards/anonymous.yml]: Any row with null can safely be deleted

pro-tip - in sublime select `: null`, then `ctrl + g` then backspace til the lines are cleared

File: server/middleware/dashboards/anonymous.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    name: "anonymous/referrers"
+    query: "page:view->count(user.ab.allocationId)->filter(!user.uuid)->group(page.referrer.host)"
+    printer: "PieChart"
+    freeze: null
+    datalabel: "Continent"
+  -
+    question: "Are anons clicking on the propensity AB test"
+    dashboardfeature: "anonymous"
+    queryname: "propensityAB/clicks"
+    name: "anonymous/propensityAB/clicks"
```
Comment (wheresrhys) [inline server/middleware/dashboards/anonymous.yml]: I think we need to think about deduping / making less verbose all this naming stuff, but perhaps wait til we have more dashboards built so we can see what the structure coudl look like

File: views/partials/chart-configurator.html [hunk 4]
```diff
@@ -3,23 +3,24 @@
...
+			<input class="o-forms-radio o-forms-radio--small" id="this_14_days_{{index}}" name="timeframe" type="radio" value="this_14_days" {{#ifEquals @root.timeframe 'this_14_days'}}checked{{/ifEquals}}><label class="o-forms-label" for="this_14_days_{{index}}">14 days</label>
+			<input class="o-forms-radio o-forms-radio--small" id="this_30_days_{{index}}" name="timeframe" type="radio" value="this_30_days" {{#ifEquals @root.timeframe 'this_30_days'}}checked{{/ifEquals}}><label class="o-forms-label" for="this_30_days_{{index}}">30 days</label>
 			<fieldset>
 				Custom date:
 				<label>start</label> <input class="chart__configurator__start" name="timeframe[start]" type="date" value="{{@root.timeframe.start}}"> |
 				<label>end</label> <input class="chart__configurator__end" name="timeframe[end]" type="date" value="{{@root.timeframe.end}}">
 			</fieldset>
 	{{/if}}
-		<select name="interval" class="chart__configurator__interval o-forms-select">
+		<select name="interval" class="chart__configurator__interval o-forms-select o-forms-select--small">
```
Comment (wheresrhys) [inline views/partials/chart-configurator.html]: good move

**Comment by wheresrhys** [comment]
Excellent stuff. Aside from the tidyup suggestions all looks good. I'll leave charts vs postulates to your discretion

File: server/controllers/dashboard.js (hunk 1 above)
Comment (adambraimbridge) [inline server/controllers/dashboard.js]: Agreed. Now it's more about beacon, and not about CLI, "chart" is appropriate.

**Pull request merged**